tests/BenchmarkDotNet.Tests/Exporters/MarkdownExporterApprovalTests.cs

```diff
@@ -262,10 +263,41 @@ public class Escape_ParamsAndArguments
             {
                 [Params("\t", "\n"), UsedImplicitly] public string StringParam;
 
-                [Arguments('\t')] [Arguments('\n')]
-                [Benchmark] public void Foo(char charArg) {}
-                [Benchmark] public void Bar() {}
+                [Arguments('\t'), Arguments('\n')]
+                [Benchmark] public void Foo(char charArg) { }
+                [Benchmark] public void Bar() { }
+            }
+
+            /* Param Arrays */
+
+            [BaselineColumn]
+            [LogicalGroupColumn]
+            public class ParamArray
+            {
+                [Params(new byte[] { 0 }, new byte[] { 1 }, new byte[] { 0 })]
```

Comment:
what do we gain by having duplicate params (`new byte[] { 0 }`)? one day we might move https://github.com/dotnet/performance/blob/main/src/harness/BenchmarkDotNet.Extensions/UniqueArgumentsValidator.cs into BDN and this would be considered as a bug

```suggestion
                [Params(new byte[] { 0 }, new byte[] { 1 })]
```

Reply:
#### You want to remove the purpose of these tests and PR :)
These tests check that arrays can be grouped if they have the same primitive values (but they are actually different arrays).

We can disallow duplicate arguments for `[Params]` because `Attribute` supports only the built-in types. (IMO it should do Roslyn Analyzer only, not a validator)

#### But how can we disable it for [ParamSource]?
If the values are not primitive, it fallbacks to `.ToString`, which means that __most of__ user classes are equal:
```C#
x.ToString().CompareTo(y.String()) == 0 //always if .ToString is not overloaded
```
#### UniqueArgumentsValidator
If I remember correctly, you cannot making a difference between [Params] and [ParamSource] values, so `UniqueArgumentsValidator` falls for:
```C#
[ArgumentSource("GetValues")]
[Benchmark]
public void Benchmark(MyType instance) { }

public IEnumerable<MyType> GetValues()
{
    // fails because both ToString() are equal:
    yield return new MyType { Value = 10 };
    yield return new MyType { Value = 100 };
}

public class MyType { public int Value; }
```

Reply:
If the type implements `IComparable` then we shouldn't jump to ToString() (because it can be overloaded).
So the code will look like:
```C#
if(IComparableComparer.TryCompareTo(x, y, var userCompareTo))
    return userCompareTo;
```
But it works for primitives/arrays.

Are you satisfied with the current implementation, or should I remake it?